File: Year 2/Semester 4/MPP/ProtobuffProjectCSharp/networking/utils/ProtobuffConcurrentServer.cs
using log4net.Repository.Hierarchy;
using log4net;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using ProtobuffNetworking;
using ProtobuffProject.services;

namespace ProtobuffProject.networking.utils
{
    public class ProtobuffConcurrentServer : AbsConcurrentServer
    {
        private readonly IServices service;
        private static readonly ILog logger = LogManager.GetLogger("RpcConcurrentServer");
      
        private ProtoWorker worker;
        public ProtobuffConcurrentServer(string host, int port, IServices service) : base(host, port)
        {
            this.service = service; 
            logger.Info("ProtobuffConcurrentServer created");
            Console.WriteLine(@"ProtobuffConcurrentServer created");
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new ProtoWorker(service, client);
            return new Thread(new ThreadStart(worker.Run));
        }

    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 10/ProtobuffProjectJavaORM/Networking/src/main/java/gRPCProject/network/utils/ProtobuffConcurrentServer.java
package gRPCProject.network.utils;

import gRPCProject.networking.protobuffprotocol.ProtoWorker;
import gRPCProject.service.IService;

import java.net.Socket;

public class ProtobuffConcurrentServer extends AbsConcurrentServer{
    private IService chatServer;
    public ProtobuffConcurrentServer(int port, IService chatServer) {
        super(port);
        this.chatServer = chatServer;
        System.out.println("Chat- ChatProtobuffConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        ProtoWorker worker=new ProtoWorker(chatServer,client);
        Thread tw=new Thread(worker);
        return tw;
    }
}
